sql/core/src/test/scala/org/apache/spark/sql/SQLQueryTestSuite.scala

```diff
@@ -83,6 +83,15 @@ class SQLQueryTestSuite extends QueryTest with SharedSQLContext {
 
   private val regenerateGoldenFiles: Boolean = System.getenv("SPARK_GENERATE_GOLDEN_FILES") == "1"
 
+  private val testFilter: Option[String] = {
+    val testFilter = System.getenv("SPARK_SQL_QUERY_TEST_FILTER")
+    if (testFilter != null && !testFilter.isEmpty) {
+      Some(testFilter.toLowerCase(Locale.ROOT))
+    } else {
+      None
+    }
+  }
+
```

Comment:
This is not related to this pr though, I think it is some useful to run tests selectively in `SQLQueryTestSuite` (cuz the number of tests there grows recently...). If possibly, could we add this feature in a separate pr? Otherwise, I'll drop this.

Reply:
Let us create a separate PR?

Reply:
ok, I'll do later.